### PR TITLE
SVG Lint test

### PR DESCRIPTION
**Pull request by Carlgo11**
I've developed a SVG lint test to check for syntax errors and minification possibilities but would like feedback on if it's something that would be desirable to use or not.

While keeping in mind the barrier to entry for new contributors I wanted some way of automate testing of SVG files and provide feedback on ways to enhance the files. I've therefore designed the script to fail on errors which would cause the graphic not to be displayed properly on the website whilst providing messages (warnings) on things which could hint at the file not being very optimized. My idea is for the errors to have to be fixed before a PR can be merged but any warnings to be treated as debug/info messages for the PR author and maintainers.

### CODE REVIEW

File: tests/svg-lint.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'json'
+require 'active_support/core_ext/hash'
```
Comment (phallobst) [inline tests/svg-lint.rb:6]:
```suggestion
require 'active_support'
require 'active_support/core_ext/hash'
```

File: tests/svg-lint.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+def warn(file, msg)
+  puts "::warning file=#{file}:: #{msg}"
+end
+
+def test(svg, css)
+  svg.css(css).length.positive?
+end
+
+@status = 0
+diff = `git diff --name-only --diff-filter=Ard -name '*.svg' origin/master...HEAD img/`
```
Comment (phallobst) [inline tests/svg-lint.rb]:
```suggestion
diff = `git diff --name-only --diff-filter=Ard origin/master...HEAD -- 'img/***.svg'`
```

**Review by phallobst — CHANGES_REQUESTED** [review]
The script did not run without these changes on my machine.

Afterwards, I got no SVG errors, but warnings for 2/3 of the icons. :fearful:

**Review by phallobst — APPROVED** [review]

**Comment by Carlgo11** [comment]
Thanks for the review @phallobst :pray: 
requirements have been fixed and the git diff command fixed.

two-thirds of the SVGs are indeed causing warnings so imagine the potential of this script :wink:

**Pull request merged**